dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java

```diff
@@ -435,50 +436,42 @@ String getURLQueryStringFromRequest(HttpServletRequest request)
-
-			if (null == user) {//Not logged in user
-
-				Logger.debug(this.getClass(),
-						"CHECKING PERMISSION: Page doesn't have anonymous access ["
-								+ requestedURIForLogging + "]");
-				Logger.debug(this.getClass(), "401 URI = " + requestedURIForLogging);
-				Logger.debug(this.getClass(), "Unauthorized URI = " + requestedURIForLogging);
-
-				request.getSession().setAttribute(
-						com.dotmarketing.util.WebKeys.REDIRECT_AFTER_LOGIN, requestedURIForLogging);
-				response.sendError(HttpServletResponse.SC_UNAUTHORIZED,
-						"The requested page/file is unauthorized");
-				return true;
-			} else if (!permissionAPI.getRolesWithPermission(permissionable, PERMISSION_READ)
-					.contains(APILocator.getRoleAPI().loadLoggedinSiteRole())) {
-				// User is logged in need to check user permissions
-				if (!permissionAPI
-						.doesUserHavePermission(permissionable, PERMISSION_READ, user,
-								true)) {
-					// the user doesn't have permissions to see this page
-					// go to unauthorized page
-					Logger.warn(this.getClass(),
-							"CHECKING PERMISSION: Page doesn't have any access for this user ["
-									+ requestedURIForLogging + "]");
-					response.sendError(HttpServletResponse.SC_FORBIDDEN,
-							"The requested page/file is forbidden");
-					return true;
-				}
-			}
-		}
+    public boolean isUnauthorizedAndHandleError(final Permissionable permissionable, final String requestedURIForLogging,
+            final User user, final HttpServletRequest request, final HttpServletResponse response)
+            throws IOException, DotDataException {
+
+        final PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+
+        PageMode mode = PageMode.get(request);
+        // Check if the page is visible by a CMS Anonymous role
+        if (!permissionAPI.doesUserHavePermission(permissionable, PERMISSION_READ, user, mode.respectAnonPerms)) {
+
+            if (null == user) {// Not logged in user
+
+                Logger.debug(this.getClass(),
+                        "CHECKING PERMISSION: Page doesn't have anonymous access [" + requestedURIForLogging + "]");
+                Logger.debug(this.getClass(), "401 URI = " + requestedURIForLogging);
+                Logger.debug(this.getClass(), "Unauthorized URI = " + requestedURIForLogging);
+
+                request.getSession().setAttribute(com.dotmarketing.util.WebKeys.REDIRECT_AFTER_LOGIN, requestedURIForLogging);
+                response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "The requested page/file is unauthorized");
+                return true;
+            } else if (!permissionAPI.getRolesWithPermission(permissionable, PERMISSION_READ)
+                .contains(APILocator.getRoleAPI().loadLoggedinSiteRole())) {
+                // User is logged in need to check user permissions
+                if (!permissionAPI.doesUserHavePermission(permissionable, PERMISSION_READ, user, true)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)